components-compose/src/main/java/uk/gov/hmrc/components/compose/molecule/selectrow/SelectRowView.kt

```diff
@@ -0,0 +1,129 @@
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.selection.selectable
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconToggleButton
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.semantics.Role
+import androidx.compose.ui.unit.dp
+import uk.gov.hmrc.components.compose.R
+import uk.gov.hmrc.components.compose.ui.theme.HmrcTheme
+
+object SelectRowView {
+
+    private const val RESTRICTED_ITEM_COUNT = 4
+    private const val INITIAL_SELECTED_ROW = -1
+
+    /**
+     * Composable function that creates a SelectRowView with icons and text.
+     *
+     * @param selectRowViewItems List of items to be displayed in the selectable rows.
+     * @param checkedIcon Resource ID for the checked icon.
+     * @param uncheckedIcon Resource ID for the unchecked icon.
+     * @param initialSelection Initial selection position of a SelectRowView
+     * @param errorText Error text to be displayed, if any.
+     * @param onRowSelected Callback function triggered when an item is clicked.
+     */
+    @Composable
+    operator fun invoke(
+        selectRowViewItems: List<String>,
+        checkedIcon: Int = R.drawable.components_select_row_circle_checked,
+        uncheckedIcon: Int = R.drawable.components_select_row_circle_unchecked,
+        initialSelection: Int = INITIAL_SELECTED_ROW,
+        errorText: String = "",
+        onRowSelected: (position: Int, value: String) -> Unit
+    ) {
+        val selectedValue = remember { mutableStateOf(initialSelection) }
+        val restrictedItems = selectRowViewItems.take(RESTRICTED_ITEM_COUNT)
+
+        Column(Modifier.padding(8.dp)) {
+
+            if (errorText.isNotEmpty()) {
+                Text(
+                    text = errorText,
+                    color = HmrcTheme.colors.hmrcRed,
+                    modifier = Modifier.offset(HmrcTheme.dimensions.hmrcSpacing8)
+                )
+                Spacer(modifier = Modifier.height(HmrcTheme.dimensions.hmrcSpacing8))
+            }
+            restrictedItems.forEachIndexed { index, selectRow ->
+                Row(
+                    verticalAlignment = Alignment.CenterVertically,
+                    modifier = Modifier
+                        .selectable(
+                            selected = (selectedValue.value == index),
+                            onClick = {
+                                selectedValue.value = index
+                                onRowSelected(index, selectRow)
+                            },
+                            role = Role.RadioButton
+                        )
+                        .padding(HmrcTheme.dimensions.hmrcSpacing8)
```

Comment:
Can we make sure the touch state ripple abides to material design principles and extends to each side of the card? At the moment, it get's cut off before the edge of the card.